libs/common-transfers/src/lib/components/excluded-dates.component.ts

```diff
@@ -49,33 +43,50 @@ export class ExcludedDatesComponent implements PaymentFormField, OnInit, OnDestr
       this.config,
     ) as FormControl;
 
-    this.control.setValidators([
-      Validators.required,
-      this.excludedDateService.excludedDateValidator(),
-      this.excludedDateService.futureDateValidator(),
-      this.excludedDateService.pastDateValidator(),
-    ]);
+    this.classes = this.options.cssClasses.length ? this.options.cssClasses : this.classes;
+    this.validationLabel = this.options.validationMessageLabel || this.validationLabel;
+    this.validationMessages = [ ...this.validationMessages, ...this.options.validationMessages ];
+
+    this.control.setAsyncValidators([
+      this.excludedDateService.excludedDateValidator()
+    ])
     this.control.updateValueAndValidity();
     this.validationControl = this.group.controls[this.config?.name] as FormControl;
     triggerHook(PaymentFormFieldHooks.onInit, this);
   }
 
-  markDisabled(ngb: NgbDate) {
-    console.log(this.validationControl, this.control);
-    const { year, month, day } = ngb;
-    const date = `${year}-${month < 10 ? '0' + month : month}-${day < 10 ? '0' + day : day}`;
-    return this.excludedDateService.isDateValid(date);
-  }
+  markDisabled$ = this.excludedDateService.isDateExcluded$.pipe(
+    map((isDateExcludedFn) => {
+      return (ngb: NgbDate) => {
+        const { year, month, day } = ngb;
+        const date = `${year}-${month < 10 ? '0' + month : month}-${day < 10 ? '0' + day : day}`;
+        return isDateExcludedFn(date);
+      }
+    })
+  )
 
-  get minDate(): NgbDateStruct {
-    const today = new Date(this.excludedDateService.startDate);
-    return { year: today.getFullYear(), month: today.getMonth() + 1, day: today.getDate() };
-  }
+  minDate$: Observable<NgbDateStruct> = this.excludedDateService.startDate$.pipe(
+    map((startDate) => {
+      let date = new Date(startDate);
 
-  get maxDate(): NgbDateStruct {
-    const today = new Date(this.excludedDateService.endDate);
-    return { year: today.getFullYear(), month: today.getMonth() + 1, day: today.getDate() };
-  }
+      const pastDateValidatorApplied = this.options.validators.filter((validator) => validator.name === 'pastDateValidator').length;
```

Comment:
There is the possibility that the omni payment config has the OOB `pastDate` validator applied. There is, therefore, a potential scenario where the `startDate` received in the response from the restricted date service is in the past, and the current date should be accepted as the minimum date in the form instead (since it is more recent)